compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -577,7 +577,14 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
                     return None;
                 }
             }
-            ProjectionElem::Downcast(name, index) => ProjectionElem::Downcast(name, index),
+            ProjectionElem::Downcast(name, index) => {
+                if let Some(ct) = self.eval_to_const(value)
```

Comment:
We don't bail on the `None` case since GVN runs on polymorphic MIR. Is there a better uninhabited method for possibly polymorphic types?